### ISSUE TITLE
Bring back support for JS/TS analysis

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/sonar-css#190]
See #188

**Comment by Godin** [comment]
I think that error described in #188

```
Error: Cannot find module 'postcss-jsx/extract'
```

is due to missing `node_modules/postcss-jsx` that was present in b4b9eb58b027d33f263fcdbf066f3440cdd44283, but after #186 in next commit ebac7cd722f1f455344f46bd3581a3ba4cfd9003 became `node_modules/stylelint/node_modules/postcss-jsx`.

According to

```
$ git checkout b4b9eb58b027d33f263fcdbf066f3440cdd44283
$ mvn clean package
$ ls -1 sonar-css-plugin/css-bundle/node_modules > before.txt

$ git checkout ebac7cd722f1f455344f46bd3581a3ba4cfd9003
$ mvn clean package
$ ls -1 sonar-css-plugin/css-bundle/node_modules > after.txt

$ diff before.txt after.txt
42d41
< clone-regexp
71d69
< execall
103d100
< html-tags
107d103
< import-lazy
133,134d128
< is-regexp
< is-supported-regexp-flag
148d141
< known-css-properties
198d190
< postcss-jsx

$ ls -1 sonar-css-plugin/css-bundle/node_modules/stylelint/node_modules
clone-regexp
execall
html-tags
import-lazy
is-regexp
known-css-properties
log-symbols
postcss-jsx
slash
```

there are more than just `postcss-jsx`, however seems that their relocation doesn't cause problems.

----

IMO most likely `package-lock.json` was not correctly updated:

```
$ git checkout b4b9eb58b027d33f263fcdbf066f3440cdd44283

$ rm -rf sonar-css-plugin/css-bundle/node_modules

$ sed -i.bak 's/10.0.1/10.1.0/g' sonar-css-plugin/css-bundle/package.json
$ rm sonar-css-plugin/css-bundle/package.json.bak

$ mvn clean package

$ git status
HEAD detached at b4b9eb5
Changes not staged for commit:
        modified:   sonar-css-plugin/css-bundle/package-lock.json
        modified:   sonar-css-plugin/css-bundle/package.json

$ git diff ebac7cd722f1f455344f46bd3581a3ba4cfd9003 | wc -l
0

$ ls -l node_modules/postcss-jsx | wc -l
ls: cannot access 'node_modules/postcss-jsx': No such file or directory
0
```

Note that in the above case `node_modules` does not exist before update of version in `package.json` and resulting `package-lock.json` identical to the current one.

However in case of update of `package.json` with existing `node_modules`:

```
$ git reset --hard b4b9eb58b027d33f263fcdbf066f3440cdd44283

$ git status
HEAD detached at b4b9eb5
nothing to commit, working tree clean

$ rm -rf sonar-css-plugin/css-bundle/node_modules

$ mvn clean package

$ sed -i.bak 's/10.0.1/10.1.0/g' sonar-css-plugin/css-bundle/package.json
$ rm sonar-css-plugin/css-bundle/package.json.bak

$ mvn clean package

$ git status
HEAD detached at b4b9eb5
Changes not staged for commit:
        modified:   sonar-css-plugin/css-bundle/package-lock.json
        modified:   sonar-css-plugin/css-bundle/package.json

$ git diff ebac7cd722f1f455344f46bd3581a3ba4cfd9003 | wc -l
1413

$ ls -l sonar-css-plugin/css-bundle/node_modules/postcss-jsx | wc -l
24

$ sha256sum sonar-css-plugin/css-bundle/package-lock.json
d60cdfcec8fb07786bde329bba176def2a6b216f19e572bba5e8da37a493d0a5  sonar-css-plugin/css-bundle/package-lock.json
```

In this above case `node_modules/postcss-jsx` exists and `package-lock.json` is identical to the newly generated:

```
$ rm sonar-css-plugin/css-bundle/package-lock.json

$ rm -rf sonar-css-plugin/css-bundle/node_modules

$ mvn clean package

$ sha256sum sonar-css-plugin/css-bundle/package-lock.json
d60cdfcec8fb07786bde329bba176def2a6b216f19e572bba5e8da37a493d0a5  sonar-css-plugin/css-bundle/package-lock.json

$ ls -l sonar-css-plugin/css-bundle/node_modules/postcss-jsx | wc -l
24
```

----

Usage of `npm install --save --save-exact stylelint@10.1.0` instead of manual change of `package.json` (recall that Maven executes `npm install`) leads to exactly the same results with above problem in case of absent `node_modules`:

```
$ git reset --hard b4b9eb58b027d33f263fcdbf066f3440cdd44283

$ rm -rf sonar-css-plugin/css-bundle/node_modules

$ cd sonar-css-plugin/css-bundle
$ ../target/node/node ../target/node/node_modules/npm/bin/npm-cli.js install --save --save-exact stylelint@10.1.0
$ cd -

$ git diff ebac7cd722f1f455344f46bd3581a3ba4cfd9003 | wc -l
0
```

```
$ git reset --hard b4b9eb58b027d33f263fcdbf066f3440cdd44283

$ rm -rf sonar-css-plugin/css-bundle/node_modules

$ cd sonar-css-plugin/css-bundle
$ ../target/node/node ../target/node/node_modules/npm/bin/npm-cli.js install
$ ../target/node/node ../target/node/node_modules/npm/bin/npm-cli.js install --save --save-exact stylelint@10.1.0
$ cd -

$ sha256sum sonar-css-plugin/css-bundle/package-lock.json
d60cdfcec8fb07786bde329bba176def2a6b216f19e572bba5e8da37a493d0a5  sonar-css-plugin/css-bundle/package-lock.json
```

----

Currently we use `npm` version `5.6.0`, compared to which latest as of today `npm` version `5.10.0` from `5.x` series and `6.10.0` from `6.x` series produce same slightly different `package-lock.json` and seem to not suffer from the above problem of absent `node_modules`:

```
$ git reset --hard b4b9eb58b027d33f263fcdbf066f3440cdd44283

$ sed -i.bak 's/<npmVersion>5.6.0/<npmVersion>5.10.0/g' sonar-css-plugin/pom.xml
$ rm sonar-css-plugin/pom.xml.bak

$ sed -i.bak 's/10.0.1/10.1.0/g' sonar-css-plugin/css-bundle/package.json
$ rm sonar-css-plugin/css-bundle/package.json.bak

$ mvn clean package

$ ls -l sonar-css-plugin/css-bundle/node_modules/postcss-jsx | wc -l
24

$ sha256sum sonar-css-plugin/css-bundle/package-lock.json
d81b3a3f530cb81cceff20fc6d2bacabc3ad8f30a82ee50a4d38ad8b7f492bde  sonar-css-plugin/css-bundle/package-lock.json


$ git reset --hard b4b9eb58b027d33f263fcdbf066f3440cdd44283

$ sed -i.bak 's/<npmVersion>5.6.0/<npmVersion>5.10.0/g' sonar-css-plugin/pom.xml
$ rm sonar-css-plugin/pom.xml.bak

$ rm -rf sonar-css-plugin/css-bundle/node_modules

$ cd sonar-css-plugin/css-bundle
$ ../target/node/node ../target/node/node_modules/npm/bin/npm-cli.js install --save --save-exact stylelint@10.1.0
$ cd -

$ sha256sum sonar-css-plugin/css-bundle/package-lock.json
d81b3a3f530cb81cceff20fc6d2bacabc3ad8f30a82ee50a4d38ad8b7f492bde  sonar-css-plugin/css-bundle/package-lock.json
```

```
$ git reset --hard b4b9eb58b027d33f263fcdbf066f3440cdd44283

$ sed -i.bak 's/<npmVersion>5.6.0/<npmVersion>6.10.0/g' sonar-css-plugin/pom.xml
$ rm sonar-css-plugin/pom.xml.bak

$ rm -rf sonar-css-plugin/css-bundle/node_modules

$ sed -i.bak 's/10.0.1/10.1.0/g' sonar-css-plugin/css-bundle/package.json
$ rm sonar-css-plugin/css-bundle/package.json.bak

$ mvn clean package

$ sha256sum sonar-css-plugin/css-bundle/package-lock.json
d81b3a3f530cb81cceff20fc6d2bacabc3ad8f30a82ee50a4d38ad8b7f492bde  sonar-css-plugin/css-bundle/package-lock.json


$ git reset --hard b4b9eb58b027d33f263fcdbf066f3440cdd44283

$ sed -i.bak 's/<npmVersion>5.6.0/<npmVersion>6.10.0/g' sonar-css-plugin/pom.xml
$ rm sonar-css-plugin/pom.xml.bak

$ rm -rf sonar-css-plugin/css-bundle/node_modules

$ cd sonar-css-plugin/css-bundle
$ ../target/node/node ../target/node/node_modules/npm/bin/npm-cli.js install --save --save-exact stylelint@10.1.0
$ cd -

$ sha256sum sonar-css-plugin/css-bundle/package-lock.json
d81b3a3f530cb81cceff20fc6d2bacabc3ad8f30a82ee50a4d38ad8b7f492bde  sonar-css-plugin/css-bundle/package-lock.json
```

**Comment by vilchik-elena** [comment]
I close this ticket as won't fix due to several reasons:
* stylelint applies the same rules to css inside js and css files. Some of the rules are not applicable (see  [this comment](https://github.com/stylelint/stylelint/issues/3410#issuecomment-399658964)) and should be tuned. Yet we think that ROI of such granular CSS analysis is low.
* stylelint mistakenly recognises JS code in JSX syntax as CSS, thus more FP
